implementation/leprechaun/src/shared/api/generic-paginated.ts

```diff
@@ -0,0 +1,26 @@
+import { applyDecorators, Type } from '@nestjs/common';
+import { ApiExtraModels, ApiOkResponse, ApiQuery, getSchemaPath } from '@nestjs/swagger';
+import { PaginatedDto } from './generic-paginated.dto';
+
+export const ApiPaginatedResponse = <TModel extends Type<any>>(model: TModel) => {
```

Comment:
```suggestion
export const ApiPaginatedResponse = <TModel extends Type>(model: TModel) => {
```

Reply:
I did `<any>` because it was previously complaining about types, but now it doesn't? huh. Sometimes my vscode does weird things and I have to reload the window for it to work.